client/state/navigation/middleware.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import {
+	NAVIGATE_TO_ROUTE,
+} from 'state/action-types';
+
+export default ( store, next, action ) => {
+	if ( NAVIGATE_TO_ROUTE === action.type ) {
+		page( action.route );
+	}
+
+	return next( action );
```

Comment:
I don't know enough about `page()` to review this but we want to make sure we don't mess ourselves up with the timing/sequence here